### DISCUSSION TITLE
Setting `COLUMN` environment variable using `monkeypatch` messes up output of run results

### DISCUSSION THREAD

**Discussion by dgard1981** [pytest-dev/pytest#11204] [General] [answered]
I have the need to capture output at a standard width. One option is to use `monkeypatch` and set the `COLUMNS` environment variable. However, doing this is (understandably) messing up the verbose output of the `pytest` run results.

```python
@pytest.fixture(scope='function')
def mock_width(monkeypatch)
    monkeypatch.setenv('COLUMNS', '120')
```

I was hoping that I would be able to use `monkeypatch.delenv()` to work around this, but it seems that the fixture finalizer is run after the test function results have been passed to the terminal, so the below has no effect.

```python
@pytest.fixture(scope='function')
def mock_width(monkeypatch) -> None:
    monkeypatch.setenv('COLUMNS', '120')
    yield
    monkeypatch.delenv('COLUMNS')
```

Is my only option to use monkeypatch.setenv() at the beginning of every test function that tests terminal output, and monkeypatch.delenv() at the end. Or is there a better way?

Thanks.

**Comment by nicoddemus** [comment, marked as answer]
> Is my only option to use monkeypatch.setenv() at the beginning of every test function that tests terminal output, and monkeypatch.delenv() at the end. Or is there a better way?

Yeah you will need to limit the scope where `COLUMNS` is set, so I suggest to write your own context manager:

```python 

@contextmanager
def using_column_width(w: int):
    mp = pytest.MonkeyPatch()
    mp.setenv("COLUMNS", w)
    try:
        yield
    finally:
        mp.undo()

def test_foo():
    with using_column_width(120): 
        ...
```

**Reply by dgard1981** [reply]
Thank you very much, that is perfect for my needs.
